jf_agent/jf_jira/jira_download.py

```diff
@@ -713,3 +713,200 @@ def _search_users(
+                if e.status_code == 403:
+                    agent_logging.log_and_print(
+                        logger,
+                        logger.ERROR,
+                        f"you do not have the required 'development field' permissions in jira required to scan for missing repos",
+                    )
+                    return []
+
+            for repo in repositories:
+                repo_name = repo['name']
+                repo_url = repo['url']
+
+                if repo_name not in missing_repositories:
+                    missing_repositories[repo_name] = {
+                        'name': repo_name,
+                        'url': repo_url,
+                        'instance_type': instance_type,
+                    }
+
+    print()
+
+    try:
+        _remove_mismatched_repos(
+            missing_repositories, _get_repos_from_git(git_connection, config), config
+        )
+    except Exception as e:
+        print(
+            f'WARNING: Got an error when trying to cross reference repos discoverd by '
+            f'Jira with Git repos: {e}\nSkipping this process and returning all repos '
+            f'discovered by Jira...'
+        )
+
+    result = [{'name': k, 'url': v['url']} for k, v in missing_repositories.items()]
+    return result
+
+
+def _get_repos_from_git(git_connection, config):
+    if config.git_provider == 'bitbucket_server':
+
+        from jf_agent.git.bitbucket_server import get_projects as get_projects_bbs
+        from jf_agent.git.bitbucket_server import get_repos as get_repos_bbs
+
+        projects = get_projects_bbs(
+            git_connection, config.git_include_projects, config.git_exclude_projects, False
+        )
+        _, repos = zip(
+            *get_repos_bbs(
+                git_connection, projects, config.git_include_repos, config.git_exclude_repos, False
+            )
+        )
+
+    elif config.git_provider == 'bitbucket_cloud':
+
+        from jf_agent.git.bitbucket_cloud_adapter import BitbucketCloudAdapter
+
+        bbc_adapter = BitbucketCloudAdapter(git_connection)
+
+        projects = bbc_adapter.get_projects(config.git_include_projects, False)
+        repos = bbc_adapter.get_repos(
+            projects, config.git_include_repos, config.git_exclude_repos, False
+        )
+
+    elif config.git_provider == 'github':
+
+        from jf_agent.git.github import get_repos as get_repos_gh
+
+        _, repos = zip(
+            *get_repos_gh(
+                git_connection,
+                config.git_include_projects,
+                config.git_include_repos,
+                config.git_exclude_repos,
+                False,
+            )
+        )
+
+    elif config.git_provider == 'gitlab':
+
+        from jf_agent.git.gitlab_adapter import GitLabAdapter
+
+        gl_adapter = GitLabAdapter(git_connection)
+
+        projects = gl_adapter.get_projects(config.git_include_projects, False)
+        repos = gl_adapter.get_repos(
```

Comment:
nit: i think you could return early for these repos. this may just be a personal reference, in which case feel free to ignore me. 
```
...
if =='':
  return gl_adapeter.get_repos()
if== ""
  return other_way_to_get_repos()
raise ValueError()
```